template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ClassLibrary-CSharp/.template.config/dotnetcli.host.json

```diff
@@ -16,6 +16,10 @@
     "langVersion": {
       "longName": "langVersion",
       "shortName": ""
+    },
```

Comment:
It's not required to add parameter to `dotnetcli.host.json` if name is the same, and we'd like to avoid to adding more changes to host file unless absolutely needed

Reply:
@jmarolf I had to add it back, as it leads to unexpected behavior.

@KathleenDollard I was not correct here yesterday, and we will need to add it here for two reasons:
1. Current parser is case sensitive, and `--Nullable` and `--nullable` are different options. Not sure if this is true for new parser(System.CommandLine). This can be fixed by changing case in `template.json`, and imo we should probably adopt naming symbols with small character in `template.json`, to avoid putting entry here just for case change, and make respective changes in other templates (for example now we have `Framework`, but `langVersion`). Alternatively we can change the case of first letter programmatically.

2. (and more important) if we don't add the entry here with `shortName` empty, the short name alias will be randomly generated for symbol (for `--Nullable` it will be `-N`; for `--nullable` it will be `-nu` as `-n` is already taken for `--name`). I'm not sure if this behavior is known, desirable and should be enabled by default. In any case I would vote not for changing it old parser, however consider changing it when we move to new parser.
Note however that this will be a breaking change for whoever used this feature before.

Reply:
Thanks folks for getting this figured out and catching the weird behavior before we merged. 

Let's move toward not requiring hosts for simple cases. Removing the "feature" of shortcut assignments will be a breaking change, so let's discuss how to incorporate that, but probably not .NET 6.

When we change the parser, we can make switches case insensitive, but this is not expected on all platforms so we should make sure this is consistent across the CLI. 

So, merge this, and check for these two items on our post .NET 6 backlog.

Reply:
Thank you, I've created https://github.com/dotnet/templating/issues/3237 and https://github.com/dotnet/templating/issues/3238 in new parser epic to follow this up. I'm merging PR now.